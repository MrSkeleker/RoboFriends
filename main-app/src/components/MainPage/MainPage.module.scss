@font-face {
	font-display: swap;
	font-family: 'SEGA LOGO FONT';
	font-style: normal;
	font-weight: normal;
	src: local('SEGA LOGO FONT'), url('../../assets/fonts/SEGA.woff') format('woff');
}

.mainPage {
	text-align: center;

	h1 {
		font-weight: 200;
		color: #0ccac4;
		font-family: 'SEGA LOGO FONT', sans-serif;
	}

	.loader {
        margin: 30px auto;
		border: 16px solid #f3f3f3; /* Light grey */
		border-top: 16px solid #3498db; /* Blue */
		border-radius: 50%;
		width: 120px;
		height: 120px;
		animation: spin 2s linear infinite;
	}
}

@keyframes spin {
	0% {
		transform: rotate(0deg);
	}
	100% {
		transform: rotate(360deg);
	}
}