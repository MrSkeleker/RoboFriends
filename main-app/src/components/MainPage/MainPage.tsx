import React, { useEffect } from 'react';

import styles from './MainPage.module.scss';

import CardsList from '../CardsList/CardsList';
import SearchBox from '../SearchBox/SearchBox';
import Scroll from '../Scroll/Scroll';
import Header from '../Header/Header';
import ErrorBoundary from '../ErrorBoundary/ErrorBoundary';

function MainPage(props: any) {
    const { robots, isPending, searchField, onSearchChange, onRequestRobots } = props;
    useEffect(() => {
        onRequestRobots();
    }, [onRequestRobots]);
    const filtredRobots = robots?.filter((robot: any) => robot.name.toLowerCase().includes(searchField.toLowerCase()));
    return (
        <div className={styles.mainPage}>
            <Header title='RoboFriends' />
            <SearchBox setSearch={onSearchChange} searchValue={searchField} />
            <Scroll>
                <ErrorBoundary>
                    {isPending
                        ? <div className={styles.loader} />
                        : <CardsList robots={filtredRobots} />}
                </ErrorBoundary>
            </Scroll>
        </div>
    );
}

export default MainPage;