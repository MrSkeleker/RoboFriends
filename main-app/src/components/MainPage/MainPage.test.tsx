import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

describe('Test MainPage Component', () => {
    let wrapper: any;
    const mockProps = {
        robots: [{
            id: -1,
            name: 'John',
            email: 'john@gmail.com'
        },
        {
            id: -2,
            name: 'Tom',
            email: 'tom@gmail.com'
        }],
        searchField: '',
        isPending: false,
        onSearchChange: jest.fn(),
        onRequestRobots: jest.fn()
    }
    beforeEach(() => {
        wrapper = shallow(<MainPage {...mockProps}/>);
    });

    it('test base component', ()=>{
        expect(wrapper).toMatchSnapshot();
    })

    it('test isPending branches', ()=>{
        const wrapper2 = shallow(<MainPage {...mockProps} isPending={true}/>);
        expect(wrapper2).toMatchSnapshot();
    })
})
