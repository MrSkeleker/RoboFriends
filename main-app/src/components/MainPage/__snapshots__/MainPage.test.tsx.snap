// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MainPage Component test base component 1`] = `
<div
  className="mainPage"
>
  <Header
    title="RoboFriends"
  />
  <SearchBox
    searchValue=""
    setSearch={[MockFunction]}
  />
  <Scroll>
    <ErrorBoundary>
      <CardsList
        robots={
          Array [
            Object {
              "email": "john@gmail.com",
              "id": -1,
              "name": "John",
            },
            Object {
              "email": "tom@gmail.com",
              "id": -2,
              "name": "Tom",
            },
          ]
        }
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;

exports[`Test MainPage Component test isPending branches 1`] = `
<div
  className="mainPage"
>
  <Header
    title="RoboFriends"
  />
  <SearchBox
    searchValue=""
    setSearch={[MockFunction]}
  />
  <Scroll>
    <ErrorBoundary>
      <div
        className="loader"
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
